src/value.cpp

```diff
@@ -133,20 +133,16 @@ DataValue* DataValue::clone_() const {
 }
 
 std::ostream& DataValue::write(std::ostream& os) const {
-  size_t end = value_.size();
-  for (size_t i = 0; i != end; ++i) {
-    os << static_cast<int>(value_.at(i));
-    if (i < end - 1)
-      os << " ";
+  if (!value_.empty()) {
+    std::copy(value_.begin(), value_.end() - 1, std::ostream_iterator<int>(os, " "));
+    os << static_cast<int>(value_.back());
```

Comment:
This seems a weird way to write the code. Is it a standard idiom that I'm not familiar with? If so, please could you add a comment with a hyperlink to the relevant wikipedia page or similar?

Reply:
I don't think it's standard. The goal way to avoid a manual loop.